torch/_refs/__init__.py

```diff
@@ -3247,10 +3251,8 @@ def index_select(x: TensorLike, dim: int, index: TensorLike):
     )
     # Treat scalars as elements of \R^1
     if x.ndim == 0:
-        # we cannot write `x.unsqueeze(0)[index].squeeze(0).clone()`
-        # as tensor[index] will trigger index.item() if index is a 0-dim tensor
-        # and .item() cannot be symbolically traced with FakeTensor.
-        return torch.ops.aten.index(x.unsqueeze(0), [index]).squeeze(0).clone()
+        return x.unsqueeze(0)[index].squeeze(0).clone()
```

Comment:
what happened to the comment that said "we can't do that"?

Reply:
This is a reference, so it cannot use `aten` ops in them. I can write it as:
```python
out = x.new_empty(index.shape)
out.index_copy_(dim, index, x)
```

Now, this is what we get from not having a proper set of ops to model advanced indexing...

Reply:
For example:
```python
torch.empty_like(x).index_copy(0, index, x.expand_as(index))
```
This is of course a terrible hack but it would just fire in the setting when `x` is 0-dim tensor, so I think that with a comment it should be fine. thoughts?

Reply:
Yeah this looks like a bug in eager, but still, that doesn't change the fact that the expression above becomes unraceable with `x.unsqueeze(0)[index]` and we shouldn't have untraceable decompositions. cc @mruberry

Reply:
What about using the proposed decomposition in https://github.com/pytorch/pytorch/pull/87183#discussion_r999454333?

Reply:
Yeah the workaround seems ok